src/daft-csv/src/options.rs

```diff
@@ -0,0 +1,299 @@
+    pyo3::{
+        pyclass, pyclass::CompareOp, pymethods, types::PyBytes, PyObject, PyResult, PyTypeInfo,
+        Python, ToPyObject,
+    },
+};
+
+/// Options for converting CSV data to Daft data.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[cfg_attr(feature = "python", pyclass(module = "daft.daft"))]
+pub struct CsvConvertOptions {
+    pub limit: Option<usize>,
+    pub include_columns: Option<Vec<String>>,
+    pub column_names: Option<Vec<String>>,
+    pub schema: Option<SchemaRef>,
+}
+
+impl CsvConvertOptions {
+    pub fn new_internal(
+        limit: Option<usize>,
+        include_columns: Option<Vec<String>>,
+        column_names: Option<Vec<String>>,
+        schema: Option<SchemaRef>,
+    ) -> Self {
+        Self {
+            limit,
+            include_columns,
+            column_names,
+            schema,
+        }
+    }
+
+    pub fn with_limit(self, limit: Option<usize>) -> Self {
+        Self {
+            limit,
+            include_columns: self.include_columns,
+            column_names: self.column_names,
+            schema: self.schema,
+        }
+    }
+
+    pub fn with_include_columns(self, include_columns: Option<Vec<String>>) -> Self {
+        Self {
+            limit: self.limit,
+            include_columns,
+            column_names: self.column_names,
+            schema: self.schema,
+        }
+    }
+
+    pub fn with_column_names(self, column_names: Option<Vec<String>>) -> Self {
+        Self {
+            limit: self.limit,
+            include_columns: self.include_columns,
+            column_names,
+            schema: self.schema,
+        }
+    }
+
+    pub fn with_schema(self, schema: Option<SchemaRef>) -> Self {
+        Self {
+            limit: self.limit,
+            include_columns: self.include_columns,
+            column_names: self.column_names,
+            schema,
+        }
+    }
+}
+
+impl Default for CsvConvertOptions {
+    fn default() -> Self {
+        Self::new_internal(None, None, None, None)
+    }
+}
+
+#[cfg(feature = "python")]
+#[pymethods]
+impl CsvConvertOptions {
+    /// Create conversion options for the CSV reader.
+    ///
+    /// # Arguments:
+    ///
+    /// * `limit` - Only read this many rows.
+    /// * `include_columns` - The names of the columns that should be kept, e.g. via a projection.
+    /// * `column_names` - The names for the CSV columns.
+    /// * `schema` - The names and dtypes for the CSV columns.
+    #[new]
+    #[pyo3(signature = (limit=None, include_columns=None, column_names=None, schema=None))]
+    pub fn new(
+        limit: Option<usize>,
+        include_columns: Option<Vec<String>>,
+        column_names: Option<Vec<String>>,
+        schema: Option<PySchema>,
+    ) -> Self {
+        Self::new_internal(
+            limit,
+            include_columns,
+            column_names,
+            schema.map(|s| s.into()),
+        )
+    }
+
+    #[getter]
```

Comment:
you can use the pyo3 `get` attribute instead i believe
https://pyo3.rs/v0.20.0/class?search=get

Reply:
You unfortunately cannot use the field-wise `get` attribute within the context of a `cfg_attr` on the struct, and we can't use the `get_all` workaround since some of the getters aren't trivial, so we need to explicitly define getters for each field.

https://github.com/PyO3/pyo3/issues/1003